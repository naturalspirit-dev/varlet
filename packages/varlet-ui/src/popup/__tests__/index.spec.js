const Popup = require('../../../cjs/popup').default
const { render } = require('@testing-library/vue')

test('test popup', async () => {
  const wrapper = render(Popup)
  console.log(wrapper)
})
