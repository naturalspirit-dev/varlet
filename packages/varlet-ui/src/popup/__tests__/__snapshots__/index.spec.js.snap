// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test popup example 1`] = `
"<div class=\\"app-type\\">弹出位置</div><button class=\\"var-button var--box var-button--normal var--flex var-button--primary var-elevation--1 mt-10\\" style=\\"width: 100%;\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">居中弹出</div>
</button><button class=\\"var-button var--box var-button--normal var--flex var-button--primary var-elevation--1 mt-10\\" style=\\"width: 100%;\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">下方弹出</div>
</button><button class=\\"var-button var--box var-button--normal var--flex var-button--primary var-elevation--1 mt-10\\" style=\\"width: 100%;\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">上方弹出</div>
</button><button class=\\"var-button var--box var-button--normal var--flex var-button--primary var-elevation--1 mt-10\\" style=\\"width: 100%;\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">左侧弹出</div>
</button><button class=\\"var-button var--box var-button--normal var--flex var-button--primary var-elevation--1 mt-10\\" style=\\"width: 100%;\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">右侧弹出</div>
</button>
<!--teleport start-->
<!--teleport end-->
<!--teleport start-->
<!--teleport end-->
<!--teleport start-->
<!--teleport end-->
<!--teleport start-->
<!--teleport end-->
<!--teleport start-->
<!--teleport end-->
<div class=\\"app-type\\">遮罩层样式</div><button class=\\"var-button var--box var-button--normal var--flex var-button--primary var-elevation--1 mt-10\\" style=\\"width: 100%;\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">遮罩层class</div>
</button>
<!--teleport start-->
<!--teleport end--><button class=\\"var-button var--box var-button--normal var--flex var-button--primary var-elevation--1 mt-10\\" style=\\"width: 100%;\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">遮罩层style</div>
</button>
<!--teleport start-->
<!--teleport end-->
<div class=\\"app-type\\">注册事件</div><button class=\\"var-button var--box var-button--normal var--flex var-button--primary var-elevation--1 mt-10\\" style=\\"width: 100%;\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">注册事件</div>
</button>
<!--teleport start-->
<!--teleport end-->"
`;

exports[`test popup show 1`] = `
"<div class=\\"container\\">
  <!--teleport start-->
  <!--teleport end-->
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 1998; display: none;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 1999;\\"></div>
      <transition-stub>
        <div class=\\"var-popup__content var-elevation--3 var-popup--center\\" style=\\"z-index: 2000; display: none;\\"> default slot content </div>
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;

exports[`test popup show 2`] = `
"<div class=\\"container\\">
  <!--teleport start-->
  <!--teleport end-->
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 2001;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 2002;\\"></div>
      <transition-stub>
        <div class=\\"var-popup__content var-elevation--3 var-popup--center\\" style=\\"z-index: 2003;\\"> default slot content </div>
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;

exports[`test popup show 3`] = `
"<div class=\\"container\\">
  <!--teleport start-->
  <!--teleport end-->
  <transition-stub>
    <div class=\\"var--box var-popup\\" style=\\"z-index: 2001; display: none;\\">
      <div class=\\"var-popup__overlay\\" style=\\"z-index: 2002;\\"></div>
      <transition-stub>
        <div class=\\"var-popup__content var-elevation--3 var-popup--center\\" style=\\"z-index: 2003; display: none;\\"> default slot content </div>
      </transition-stub>
    </div>
  </transition-stub>
</div>"
`;
